pages/api/wallet/hashes/[address].ts

```diff
@@ -62,36 +63,44 @@ export default async function handler(
     const currentBlockNumber = await etherscanProvider.getBlockNumber();
 
     const hashes: WalletHash[] = [];
+
+    const generatedFilter = hashesContract.filters.Generated();
+    const allGeneratedEvents = await hashesContract.queryFilter(generatedFilter);
+
+    const tokenIdPromises = Array.from(Array(hashesCount), (_, i) => i).map(i => hashesContract.tokenOfOwnerByIndex(address, i));
```

Comment:
This feels a bit odd here, could you speak a bit more to what this is doing? Appears to be creating an array from 2 separate areas, but perhaps this could be written a bit simpler; would like to get context before giving feedback.